src/code-templates/libraries/redis/index.ts

```diff
@@ -0,0 +1,87 @@
+import { RedisClientType } from '@redis/client'
+import { ConnectionState, create, RedisOptions } from './redis-client'
+import { RedisResult } from './redis-result'
+
+let client: RedisClientType
+
+let state: ConnectionState = ConnectionState.DISCONNECTED
+
+export const init = async (options: RedisOptions) => {
+  if (client == null) {
+    client = create(options) as RedisClientType
+
+    await client.connect()
+    state = ConnectionState.CONNECTED
+    client.on('error', (err) => {
+      console.log('err', err)
+      state = ConnectionState.ERROR
+    })
+    client.on('ready', () => {
+      console.log('ready')
+      state = ConnectionState.CONNECTED
+    })
+  }
+}
+
+export const set = async (
+  key: string,
+  value: string,
+  ttlSeconds?: number
+): Promise<RedisResult> => {
+  if (state !== ConnectionState.CONNECTED) {
+    return { error: new Error('Connection Error') }
```

Comment:
why should we use `return` instead of `throw`? this opens for situations where you may forget to add error handling